src/analysis/processing/pdal/qgsalgorithmpdalreproject.cpp

```diff
@@ -0,0 +1,90 @@
+/***************************************************************************
+                         qgsalgorithmpdalreproject.cpp
+                         ---------------------
+    begin                : February 2023
+    copyright            : (C) 2023 by Alexander Bruy
+    email                : alexander dot bruy at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmpdalreproject.h"
+
+#include "qgsrunprocess.h"
+#include "qgspointcloudlayer.h"
+
+///@cond PRIVATE
+
+QString QgsPdalReprojectAlgorithm::name() const
+{
+  return QStringLiteral( "reproject" );
+}
+
+QString QgsPdalReprojectAlgorithm::displayName() const
+{
+  return QObject::tr( "Reproject" );
+}
+
+QString QgsPdalReprojectAlgorithm::group() const
+{
+  return QObject::tr( "Point cloud data management" );
+}
+
+QString QgsPdalReprojectAlgorithm::groupId() const
+{
+  return QStringLiteral( "pointclouddatamanagement" );
+}
+
+QStringList QgsPdalReprojectAlgorithm::tags() const
+{
+  return QObject::tr( "assign,set,transform,reproject,crs,srs" ).split( ',' );
+}
+
+QString QgsPdalReprojectAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "This algorithm reprojects point cloud to a different coordinate reference system." );
+}
+
+QgsPdalReprojectAlgorithm *QgsPdalReprojectAlgorithm::createInstance() const
+{
+  return new QgsPdalReprojectAlgorithm();
+}
+
+void QgsPdalReprojectAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterPointCloudLayer( QStringLiteral( "INPUT" ), QObject::tr( "Input layer" ) ) );
+  addParameter( new QgsProcessingParameterCrs( QStringLiteral( "CRS" ), QObject::tr( "Target CRS" ), QStringLiteral( "EPSG:4326" ) ) );
```

Comment:
This should also have a QgsProcessingParameterCoordinateOperation parameter, like QgsTransformAlgorithm

Reply:
Done